python-main.js

```diff
@@ -1075,31 +1075,61 @@ function web_editor(config) {
+                if (serial){
+                  doSerial();
+                }
+            })
+            .catch(function(err) {
+                    errorHandler(err);
+            });
+    
+        // WebUSB Disconnect Events
+        navigator.usb.addEventListener('disconnect', event => {
+            var error = {"message": "Device disconnected."};
+            errorHandler(error);
+        });
 
-            if (serial){
-              doSerial();
-            }
-        })
-        .catch(function(err) {
-            console.log("Error connecting: " + err);
-            $("#flashing-overlay-container").css("display", "flex");
-            $("#flashing-info").addClass('hidden');
+    }
 
-            // If micro:bit does not support dapjs
-            if (err.message === "No valid interfaces found."){
-                $("#flashing-overlay-error").html('<div>' + err + '</div><div>'+ config["translate"]["webusb"]["err"]["update-req"] +'</div><a href="#" onclick="flashErrorClose()">'+ config["translate"]["webusb"]["close"] +'</a>');
-                return;
-            } else if (err.message === "Unable to claim interface.") {
-                $("#flashing-overlay-error").html('<div>'+ config["translate"]["webusb"]["err"]["clear-connect"] +'</div><a href="#" onclick="flashErrorClose()">'+ config["translate"]["webusb"]["close"] +'</a>');
-                return;
-            }
+    function errorHandler(err) {
+        // Disconnect
+        doDisconnect();
+
+        // Error handler
+        console.log("Error connecting: " + err);
+        $("#flashing-overlay-container").css("display", "flex");
+        $("#flashing-info").addClass('hidden');
+
+
+        // If micro:bit does not support dapjs
+        var errorMessage;
+        if (err.message === "No valid interfaces found."){
+            errorMessage = "update-req";
+        } else if (err.message === "Unable to claim interface.") {
+            errorMessage = "clear-connect";
+        } else {
+            errorMessage = "reconnect-microbit";
+        }
 
-            $("#flashing-overlay-error").html('<div>' + err + '</div><div>'+ config["translate"]["webusb"]["err"]["restart-microbit"] +'</div><a href="#" onclick="flashErrorClose()">'+ config["translate"]["webusb"]["close"] +'</a>');
+        // Show error message
+        $("#flashing-overlay-error").html(
+                '<div>' + err.message + '</div><div>' + 
+                config["translate"]["webusb"]["err"][errorMessage] +
+                '</div><a href="#" id="flashing-overlay-download">' +
+                config["translate"]["webusb"]["download"] + 
+                '</a> | <a href="#" onclick="flashErrorClose()">' +
+                config["translate"]["webusb"]["close"] +
+                '</a>'
+        );
+
+        // Attach download handler
+        $("#flashing-overlay-download").click(function () {
```

Comment:
Would `$("#flashing-overlay-download").click(doDownload);` work as well in this case?